File: src/TmrOverlay.Core/Overlays/Abstractions/OverlayDefinition.cs
namespace TmrOverlay.Core.Overlays;

internal enum OverlayContextRequirement
{
    AnyTelemetry,
    LocalPlayerInCar,
    LocalPlayerInCarOrPit
}

internal sealed record OverlayDefinition(
    string Id,
    string DisplayName,
    int DefaultWidth,
    int DefaultHeight,
    IReadOnlyList<OverlaySettingsOptionDescriptor>? Options = null,
    bool ShowSessionFilters = true,
    bool ShowScaleControl = true,
    bool ShowOpacityControl = true,
    bool FadeWhenLiveTelemetryUnavailable = false,
    OverlayContextRequirement ContextRequirement = OverlayContextRequirement.AnyTelemetry)
{
    public IReadOnlyList<OverlaySettingsOptionDescriptor> SettingsOptions { get; } = Options ?? [];
}

File: local-mac/TmrOverlayMac/Sources/TmrOverlayMac/Overlays/Abstractions/OverlayDefinition.swift
import AppKit

struct OverlayDefinition {
    let id: String
    let displayName: String
    let defaultSize: NSSize
    let showSessionFilters: Bool
    let showScaleControl: Bool
    let showOpacityControl: Bool
    let fadeWhenLiveTelemetryUnavailable: Bool

    init(
        id: String,
        displayName: String,
        defaultSize: NSSize,
        showSessionFilters: Bool = true,
        showScaleControl: Bool = true,
        showOpacityControl: Bool = true,
        fadeWhenLiveTelemetryUnavailable: Bool = false
    ) {
        self.id = id
        self.displayName = displayName
        self.defaultSize = defaultSize
        self.showSessionFilters = showSessionFilters
        self.showScaleControl = showScaleControl
        self.showOpacityControl = showOpacityControl
        self.fadeWhenLiveTelemetryUnavailable = fadeWhenLiveTelemetryUnavailable
    }
}
